### ISSUE TITLE
Attribute Error while running "bash" plugin 

### ISSUE THREAD

**Issue by raviteja7** [google/rekall#195]
Hello

I am receiving this error when running bash plugin. 

(venv) root@vps325778:~# rekal -f vpsmemory --debug -v
2016-10-11 18:53:29,829:DEBUG:rekall.1:Logging level set to 10
2016-10-11 18:53:29,831:DEBUG:rekall.1:Running plugin (shell) with args (()) kwargs ({'profile': None})

----------------------------------------------------------------------------
The Rekall Digital Forensic/Incident Response framework 1.5.3.post1.dev6 (Furka).

"We can remember it for you wholesale!"

This program is free software; you can redistribute it and/or modify it under
the terms of the GNU General Public License.

See http://www.rekall-forensic.com/docs/Manual/tutorial.html to get started.
----------------------------------------------------------------------------
[1] vpsmemory 18:53:32> bash
2016-10-11 18:53:32,677:DEBUG:rekall.1:Voting round with base: None
2016-10-11 18:53:32,678:DEBUG:rekall.1:Trying <class 'rekall.plugins.response.common.APIDummyPhysicalAddressSpace'>
2016-10-11 18:53:32,678:DEBUG:rekall.1:Failed instantiating APIDummyPhysicalAddressSpace: Instantiation failed for unspecified reason
2016-10-11 18:53:32,678:DEBUG:rekall.1:Trying <class 'rekall.plugins.addrspaces.vmem.VMSSAddressSpace'>
2016-10-11 18:53:32,679:DEBUG:rekall.1:Failed instantiating VMSSAddressSpace: No base address space provided
2016-10-11 18:53:32,679:DEBUG:rekall.1:Trying <class 'rekall.plugins.addrspaces.macho.MACHOCoreDump'>
2016-10-11 18:53:32,679:DEBUG:rekall.1:Failed instantiating MACHOCoreDump: Must stack on another address space
2016-10-11 18:53:32,679:DEBUG:rekall.1:Trying <class 'rekall.plugins.addrspaces.vmem.VMemAddressSpace'>
2016-10-11 18:53:32,679:DEBUG:rekall.1:Failed instantiating VMemAddressSpace: No base address space provided
2016-10-11 18:53:32,679:DEBUG:rekall.1:Trying <class 'rekall.plugins.addrspaces.ewf.EWFAddressSpace'>
2016-10-11 18:53:32,683:DEBUG:rekall.1:Failed instantiating EWFAddressSpace: No base address space provided
2016-10-11 18:53:32,683:DEBUG:rekall.1:Trying <class 'rekall.plugins.addrspaces.crash.WindowsCrashBMP'>
2016-10-11 18:53:32,683:DEBUG:rekall.1:Failed instantiating WindowsCrashBMP: Must stack on another address space
2016-10-11 18:53:32,683:DEBUG:rekall.1:Trying <class 'rekall.plugins.addrspaces.elfcore.KCoreAddressSpace'>
2016-10-11 18:53:32,684:DEBUG:rekall.1:Failed instantiating KCoreAddressSpace: Must stack on another address space
2016-10-11 18:53:32,684:DEBUG:rekall.1:Trying <class 'rekall.plugins.addrspaces.crash.WindowsCrashDumpSpace32'>
2016-10-11 18:53:32,684:DEBUG:rekall.1:Failed instantiating WindowsCrashDumpSpace32: No base address space provided
2016-10-11 18:53:32,684:DEBUG:rekall.1:Trying <class 'rekall.plugins.addrspaces.crash.WindowsCrashDumpSpace64'>
2016-10-11 18:53:32,684:DEBUG:rekall.1:Failed instantiating WindowsCrashDumpSpace64: No base address space provided
2016-10-11 18:53:32,684:DEBUG:rekall.1:Trying <class 'rekall.plugins.addrspaces.elfcore.Elf64CoreDump'>
2016-10-11 18:53:32,685:DEBUG:rekall.1:Failed instantiating Elf64CoreDump: Must stack on another address space
2016-10-11 18:53:32,685:DEBUG:rekall.1:Trying <class 'rekall.plugins.addrspaces.lime.LimeAddressSpace'>
2016-10-11 18:53:32,685:DEBUG:rekall.1:Failed instantiating LimeAddressSpace: Must be layered on another address space.
2016-10-11 18:53:32,685:DEBUG:rekall.1:Trying <class 'rekall.plugins.addrspaces.aff4.AFF4AddressSpace'>
2016-10-11 18:53:32,713:INFO:rekall.1:Added aff4://fed3f797-13a0-4f6b-aab9-f5622a5b11ff/PhysicalMemory as physical memory
2016-10-11 18:53:32,715:DEBUG:rekall.1:Succeeded instantiating AFF4AddressSpace
2016-10-11 18:53:32,715:DEBUG:rekall.1:Voting round with base: AFF4AddressSpace
2016-10-11 18:53:32,716:DEBUG:rekall.1:Trying <class 'rekall.plugins.response.common.APIDummyPhysicalAddressSpace'>
2016-10-11 18:53:32,716:DEBUG:rekall.1:Failed instantiating APIDummyPhysicalAddressSpace: Instantiation failed for unspecified reason
2016-10-11 18:53:32,716:DEBUG:rekall.1:Trying <class 'rekall.plugins.addrspaces.vmem.VMSSAddressSpace'>
2016-10-11 18:53:32,719:DEBUG:rekall.1:Failed instantiating VMSSAddressSpace: Invalid VMware signature: 0x0
2016-10-11 18:53:32,719:DEBUG:rekall.1:Trying <class 'rekall.plugins.addrspaces.macho.MACHOCoreDump'>
2016-10-11 18:53:32,719:DEBUG:rekall.1:Failed instantiating MACHOCoreDump: Header signature invalid
2016-10-11 18:53:32,719:DEBUG:rekall.1:Trying <class 'rekall.plugins.addrspaces.vmem.VMemAddressSpace'>
2016-10-11 18:53:32,719:DEBUG:rekall.1:Failed instantiating VMemAddressSpace: Only VMEM files supported.
2016-10-11 18:53:32,719:DEBUG:rekall.1:Trying <class 'rekall.plugins.addrspaces.ewf.EWFAddressSpace'>
2016-10-11 18:53:32,720:DEBUG:rekall.1:Failed instantiating EWFAddressSpace: EWF signature not present
2016-10-11 18:53:32,720:DEBUG:rekall.1:Trying <class 'rekall.plugins.addrspaces.crash.WindowsCrashBMP'>
2016-10-11 18:53:32,721:DEBUG:rekall.1:Failed instantiating WindowsCrashBMP: Header signature invalid
2016-10-11 18:53:32,721:DEBUG:rekall.1:Trying <class 'rekall.plugins.addrspaces.elfcore.KCoreAddressSpace'>
2016-10-11 18:53:32,721:DEBUG:rekall.1:Failed instantiating KCoreAddressSpace: Header signature invalid
2016-10-11 18:53:32,721:DEBUG:rekall.1:Trying <class 'rekall.plugins.addrspaces.crash.WindowsCrashDumpSpace32'>
2016-10-11 18:53:32,721:DEBUG:rekall.1:Failed instantiating WindowsCrashDumpSpace32: Header signature invalid
2016-10-11 18:53:32,722:DEBUG:rekall.1:Trying <class 'rekall.plugins.addrspaces.crash.WindowsCrashDumpSpace64'>
2016-10-11 18:53:32,722:DEBUG:rekall.1:Failed instantiating WindowsCrashDumpSpace64: Header signature invalid
2016-10-11 18:53:32,722:DEBUG:rekall.1:Trying <class 'rekall.plugins.addrspaces.elfcore.Elf64CoreDump'>
2016-10-11 18:53:32,722:DEBUG:rekall.1:Failed instantiating Elf64CoreDump: Header signature invalid
2016-10-11 18:53:32,722:DEBUG:rekall.1:Trying <class 'rekall.plugins.addrspaces.lime.LimeAddressSpace'>
2016-10-11 18:53:32,722:DEBUG:rekall.1:Failed instantiating LimeAddressSpace: Invalid Lime header signature
2016-10-11 18:53:32,723:DEBUG:rekall.1:Trying <class 'rekall.plugins.addrspaces.aff4.AFF4AddressSpace'>
2016-10-11 18:53:32,723:DEBUG:rekall.1:Failed instantiating AFF4AddressSpace: Must stack on another address space
2016-10-11 18:53:32,723:DEBUG:rekall.1:Trying <class 'rekall.plugins.addrspaces.pmem.MacPmemAddressSpace'>
2016-10-11 18:53:32,724:DEBUG:rekall.1:Failed instantiating MacPmemAddressSpace: Must be mapped directly over a raw device.
2016-10-11 18:53:32,724:DEBUG:rekall.1:Trying <class 'rekall.plugins.addrspaces.standard.FileAddressSpace'>
2016-10-11 18:53:32,724:DEBUG:rekall.1:Failed instantiating FileAddressSpace: Must be first Address Space
2016-10-11 18:53:32,724:DEBUG:rekall.1:Trying <class 'rekall.plugins.addrspaces.mmap_address_space.MmapFileAddressSpace'>
2016-10-11 18:53:32,724:DEBUG:rekall.1:Failed instantiating MmapFileAddressSpace: Must be first Address Space
2016-10-11 18:53:32,725:DEBUG:rekall.1:Trying <class 'rekall.plugins.addrspaces.amd64.VTxPagedMemory'>
2016-10-11 18:53:32,726:DEBUG:rekall.1:Failed instantiating VTxPagedMemory: No more EPTs specified
2016-10-11 18:53:32,726:DEBUG:rekall.1:Trying <class 'rekall.plugins.addrspaces.standard.GlobalOffsetAddressSpace'>
2016-10-11 18:53:32,726:DEBUG:rekall.1:Failed instantiating GlobalOffsetAddressSpace: File offset not specified.
2016-10-11 18:53:32,727:DEBUG:rekall.1:Trying <class 'rekall.plugins.addrspaces.pagefile.PagefilePhysicalAddressSpace'>
2016-10-11 18:53:32,727:DEBUG:rekall.1:Failed instantiating PagefilePhysicalAddressSpace: Pagefile not specified
2016-10-11 18:53:32,727:INFO:rekall.1:Autodetected physical address space AFF4AddressSpace
2016-10-11 18:53:32,735:DEBUG:rekall.1:Will detect profile using these Detectors: linux_index,nt_index,tsk,osx,pe,windows_kernel_file,rsds,ntfs,linux
2016-10-11 18:53:33,759:DEBUG:rekall.1:Opened url https://github.com/google/rekall-profiles/raw/master/v1.0/inventory.gz
2016-10-11 18:53:35,256:DEBUG:rekall.1:Opened url https://github.com/google/rekall-profiles/raw/master/v1.0/Linux/index.gz
2016-10-11 18:53:35,591:INFO:rekall.1:Loaded profile Linux/index from URL:https://github.com/google/rekall-profiles/raw/master (in 1.45575404167 sec)
2016-10-11 18:53:36,782:DEBUG:rekall.1:Opened url https://github.com/google/rekall-profiles/raw/master/v1.0/nt/eprocess_index.gz
2016-10-11 18:53:37,233:INFO:rekall.1:Loaded profile nt/eprocess_index from URL:https://github.com/google/rekall-profiles/raw/master (in 1.64005184174 sec)
2016-10-11 18:53:38,349:DEBUG:rekall.1:Opened url https://github.com/google/rekall-profiles/raw/master/v1.0/nt/index.gz
2016-10-11 18:53:38,933:INFO:rekall.1:Loaded profile nt/index from URL:https://github.com/google/rekall-profiles/raw/master (in 1.70025110245 sec)
2016-10-11 18:53:40,065:DEBUG:rekall.1:Opened url https://github.com/google/rekall-profiles/raw/master/v1.0/OSX/index.gz
2016-10-11 18:53:40,066:INFO:rekall.1:Loaded profile OSX/index from URL:https://github.com/google/rekall-profiles/raw/master (in 1.13226222992 sec)
2016-10-11 18:53:41,186:DEBUG:rekall.1:Opened url https://github.com/google/rekall-profiles/raw/master/v1.0/pe.gz
2016-10-11 18:53:41,221:INFO:rekall.1:Loaded profile pe from URL:https://github.com/google/rekall-profiles/raw/master (in 1.1537270546 sec)
2016-10-11 18:53:41,222:DEBUG:rekall.1:Trying method pe, offset 0
2016-10-11 18:53:41,224:DEBUG:rekall.1:Trying method windows_kernel_file, offset 0
2016-10-11 18:53:41,225:DEBUG:rekall.1:Trying method rsds, offset 0
2016-10-11 18:53:41,226:DEBUG:rekall.1:Trying method linux_index, offset 0
2016-10-11 18:53:41,227:DEBUG:rekall.1:LinuxIndexDetector:DetectFromHit(0) = None
2016-10-11 18:53:41,254:DEBUG:rekall.1:Found /proc/kallsyms of size: 4415077
2016-10-11 18:53:41,824:INFO:rekall.1:LinuxIndexDetector found profile Linux/Ubuntu/16.04/x86_64/4.4.0-22-generic with 6/6 matches.
2016-10-11 18:53:43,638:DEBUG:rekall.1:Opened url https://github.com/google/rekall-profiles/raw/master/v1.0/Linux/Ubuntu/16.04/x86_64/4.4.0-22-generic.gz
2016-10-11 18:53:45,518:INFO:rekall.1:Loaded profile Linux/Ubuntu/16.04/x86_64/4.4.0-22-generic from URL:https://github.com/google/rekall-profiles/raw/master (in 3.69332790375 sec)
2016-10-11 18:53:45,520:INFO:rekall.1:Found slide 0x0
2016-10-11 18:53:45,521:DEBUG:rekall.1:Calculating page offset...
2016-10-11 18:53:45,522:DEBUG:rekall.1:Calculating page offset...
2016-10-11 18:53:45,541:INFO:rekall.1:Detection method linux_index yielded profile <AMD64 profile Linux/Ubuntu/16.04/x86_64/4.4.0-22-generic (Linux)>
----------------------> bash()
2016-10-11 18:53:45,546:DEBUG:rekall.1:Running plugin (bash) with args (()) kwargs ({})
-        Timestamp         Command
  ------------------------ -------
2016-10-11 18:53:45,685:WARNING:rekall.1:vm_flags has no offset in object vm_area_struct. Check that vtypes has a concrete definition for it.
2016-10-11 18:53:45,686:CRITICAL:rekall.1:Traceback (most recent call last):
  File "/root/rekall/rekall-core/rekall/session.py", line 859, in RunPlugin
    result = plugin_obj.render(ui_renderer) or plugin_obj
  File "/root/rekall/rekall-core/rekall/plugin.py", line 720, in render
    for row in self.collect():
  File "/root/rekall/rekall-core/rekall/plugins/linux/bash.py", line 166, in collect
    timestamps = self.get_timestamps(timestamp_scanner)
  File "/root/rekall/rekall-core/rekall/plugins/linux/bash.py", line 146, in get_timestamps
    for hit in scanner.scan():
  File "/root/rekall/rekall-core/rekall/plugins/linux/common.py", line 333, in scan
    start = max(vma.vm_start, self.task.mm.start_brk)
  File "/root/rekall/rekall-core/rekall/obj.py", line 1433, in __getattr__
    raise AttributeError(attr)
AttributeError: vm_start

> /root/rekall/rekall-core/rekall/obj.py(1433)__getattr__()
-> raise AttributeError(attr)
(Pdb)

Thanks

**Comment by scudette** [comment]
Thanks for this bug report! This bug should have been fixed by 25b6c7309cd79553c8a7dd6bb5dcc77014e05802 . I will rebuild all the broken profiles now but you can just build that specific one by download the [precompiled kernel module zip file](https://github.com/google/rekall-profiles/tree/gh-pages/v1.0/src/Linux/Ubuntu/16.04/x86_64) and running

```
rekal convert_profile 4.4.0-22-generic.zip 4.4.0-22-generic.json
```

**Comment by raviteja7** [comment]
Thank you. It works now

![image](https://cloud.githubusercontent.com/assets/2549509/19289996/5f1c1d96-8fdb-11e6-8ce2-872698ccdc70.png)
